### ISSUE TITLE
Default P dynamic marking is off-center with whole notes & downstem quarter notes

### ISSUE THREAD

**Issue by DanielParker72** [musescore/MuseScore#18615]
### Issue type

Choose option...

### Bug description

The default placement of p dynamic marking is off-center with whole notes and downstem quarter notes. I did not investigate any other combination of dynamic markings and notes. Please check the screenshot, the top stave is the MuseScore 4 default, the next stave down are my corrections with a left -0.7 offset (my opinion, for reference). 

### Steps to reproduce

1. Enter a whole note
2. apply p dynamic marking
3. Enter a downstem quarter note
4. apply a p dynamic marking

### Screenshots/Screen recordings

![361220707_275530578494988_5450935793933001703_n](https://github.com/musescore/MuseScore/assets/139315507/f7f73b38-5186-41b3-9c34-aec2ca72fe7e)

### MuseScore Version

4.1.0

### Regression

I don't know

### Operating system

Ventura 13.4.1

### Additional context

I opened a brand new project and reset all style settings.

**Comment by MarcSabatella** [comment]
Those are not the default style settings - one can tell just by looking at the shape of the notes.  Using the defaults, I get this, which seems perfectly correct to me:

![image](https://github.com/musescore/MuseScore/assets/1799936/35522064-002e-4df5-97fb-179eff96778a)

I tried some non-default fonts, and it seems you customized your score to use Emmentaler.  I can confirm the odd whole note result on that font.  The quarter note also looks like yours, but to me that's better than with your adjustment - it's *supposed* to be centered under the notehead, not the stem.  At most around a -0.2 or -0.3 sp adjustment would be appropriate.  But in any case, it does indeed seem the positioning of dynamics is optimal only with Leland.

**Comment by oktophonie** [comment]
Leland gets no special treatment here. If the font metadata includes `opticalCenter` for the dynamics then that point is centered on the middle of the notehead. If there is no such metadata, the centre of the bounding box of the glyph is used.

Leland has this data, and it's possible that Emmentaler doesn't (I don't recall offhand). It looks to me like the bounding box of the p is centered on a quarter more or less as expected (it does look a little bit too far to the right given the italic style of the symbol); the stem direction makes no difference here. However, something weird is definitely going on with the whole note.

**Comment by DanielParker72** [comment]
Sorry, I forgot to write that I use the Emmentaler music font. Thank you Mark Sabatella, and hello oktophonie, nice to meet you :).

**Comment by MarcSabatella** [comment]
Indeed, I didn't mean to imply there was special treatment for Leland; presumably it's an issue with either the metadata for the Emmentaler glyphs.  Also maybe the two Finale fonts, which seem similarly off.

But, the odd thing to me though is that whatever is going on with those fonts, they don't seem to be centering on anything at all - the position of the dynamic seems to be fixed to *right* as the notehead width increases:

![image](https://github.com/musescore/MuseScore/assets/1799936/141676de-9333-4306-8d6b-136e90e06f49)

No idea what that's about.
